server/server.go

```diff
@@ -3,40 +3,114 @@ package chatbackend
 	// Create new gorillaRouter
 	gorillaRouter := gorillaMux.NewRouter()
 	// Bind an endpoint path to handleWebSocketConnection
 	gorillaRouter.HandleFunc("/chat", HandleWebSocketConnection)
+	gorillaRouter.HandleFunc("/profile", HandleProfile)
 	// Listen on a specific port for incoming connections
 	if listenAndServeErr := http.ListenAndServe(":8080", gorillaRouter); listenAndServeErr != nil {
 		// If there is an error while setting up, panic and show us the error
-		log.Fatal("ListenAndServe: " + listenAndServeErr.Error())
+		logError(syslog.LOG_CRIT, listenAndServeErr)
 	} // if listenAndServeErr
 } // func startWebSocketServer
 
+// HandleProfile decides what happens when a client requests or uploads a profile
+func HandleProfile(serverHTTPResponse http.ResponseWriter, clientHTTPRequest *http.Request) {
+	// Make sure the client passes the bearer authentication
+	if clientHTTPRequest.Header.Get("Bearer") != "5d41402abc4b2a76b9719d911017c592" {
+		http.Error(serverHTTPResponse, "Forbidden", 403)
+		return
+	} // if clientHTTPRequest.Header.Get("Bearer")
+	// Choose different actions depending on the type of request
+	switch clientHTTPRequest.Method {
+	case "GET":
+		// Get the identity public key for the client that we want to obtain the profile for
+		profile, profileErr := getProfileFromStorage(clientHTTPRequest.Header.Get("Identity"))
+		// If there is an error while obtaining the profile, inform the client
+		if profileErr != nil {
+			http.Error(serverHTTPResponse, profileErr.Error(), 500)
+			return
+		} // if profileErr != nil {
+		// Write the protobyfBytes of the Profile protobuf structure to the client, in case of ann error inform the client
+		if _, serverHTTPResponseErr := serverHTTPResponse.Write(profile); serverHTTPResponseErr != nil {
+			logError(syslog.LOG_ERR, serverHTTPResponseErr)
+			http.Error(serverHTTPResponse, serverHTTPResponseErr.Error(), 500)
+			return
+		} // if _, serverHTTPResponseErr
+	case "PUT":
+		var profileProtobuf backendProtobuf.Profile
+		// Read a base64 representation of the Profile protobuf structure bytes
+		profileBase64Bytes, readErr := ioutil.ReadAll(clientHTTPRequest.Body)
+		if readErr != nil {
+			http.Error(serverHTTPResponse, readErr.Error(), 500)
+			return
+		} // if readErr != nil {
+		// Create a string representation of the profileBase64Bytes for easier usage and decoding
+		profileBase64String := string(profileBase64Bytes)
```

Comment:
Maybe you can move this into the `base64.StdEncoding.DecodeString`. Then we use less lines

Reply:
Here I created the `profileBase64String` variable on purpose as below I use it again, not only in :
>profileProtobufBytes, profileBytesErr := base64.StdEncoding.DecodeString(`profileBase64String`)

But also in :

>persistProfileToStorageErr := persistProfileToStorage(clientHTTPRequest.Header.Get("Identity"), `profileBase64String`)

I made this tradeoff, as intuitively, calling the string(profileBase64Bytes) twice is more expensive rather than having one extra line for a variable assignment?

Reply:
ah I see. Never mind then